editor/header/tools/publish-button.js

```diff
@@ -8,81 +8,119 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from 'i18n';
+import { Component } from 'element';
 import { Button } from 'components';
 
 /**
  * Internal dependencies
  */
-import { editPost, savePost } from '../../actions';
+import {
+	editPost,
+	savePost,
+	successNotice,
+	errorNotice,
+} from '../../actions';
 import {
 	isSavingPost,
 	isEditedPostPublished,
 	isEditedPostBeingScheduled,
 	getEditedPostVisibility,
 	isEditedPostSaveable,
 	isEditedPostPublishable,
+	getCurrentPost,
+	didPostSaveRequestSucceed,
 } from '../../selectors';
 
-function PublishButton( {
-	isSaving,
-	isPublished,
-	onStatusChange,
-	onSave,
-	isBeingScheduled,
-	visibility,
-	isPublishable,
-	isSaveable,
-} ) {
-	const buttonEnabled = ! isSaving && isPublishable && isSaveable;
-	let buttonText;
-	if ( isPublished ) {
-		buttonText = __( 'Update' );
-	} else if ( isBeingScheduled ) {
-		buttonText = __( 'Schedule' );
-	} else {
-		buttonText = __( 'Publish' );
+class PublishButton extends Component {
+	constructor() {
+		super( ...arguments );
+		this.onClick = this.onClick.bind( this );
 	}
-	let publishStatus = 'publish';
-	if ( isBeingScheduled ) {
-		publishStatus = 'future';
-	} else if ( visibility === 'private' ) {
-		publishStatus = 'private';
+
+	componentDidUpdate( prevProps ) {
```

Comment:
I like the pattern of notices being an effect of actions, and just rendered by the main editor component, instead of being handled by something like the publish button.